### DISCUSSION TITLE
RDP Connections/Session visualisation

### DISCUSSION THREAD

**Discussion by fastlorenzo** [WithSecureLabs/chainsaw#140] [Show and tell]
Hi all, I've made a (quick and dirty) web app that allows to visualize RDP events by parsing the JSON output from Chainsaw: https://github.com/fastlorenzo/rdpvis

Any feedback is welcome!

**Comment by alexkornitzer** [comment]
Ah very nice, will it also execute chainsaw server side?

**Reply by fastlorenzo** [reply]
This is not something I thought about yet, it came from an incident we tried to analyze and build a timeline of RDP connections.
Although, this could be a nice feature if you want to, for example, upload the evtx file to the app and then get a timeline back.
But how it works now allows to have the data only present client side, which is great for data privacy/confidentially :smile: (and doesn't require a specific backend, just drop the HTML/JS/CSS files on a web server, or even open with your web browser, and you're done!)

**Reply by alexkornitzer** [reply]
Would you want any of this in the chainsaw cli or not really?

**Reply by fastlorenzo** [reply]
not sure I'm getting your question fully there, having the GUI in the cli to be able to generate it?

**Reply by alexkornitzer** [reply]
Yep sorry, what I meant was would having some sort of timeline generator in chainsaw's output be useful? Or does all the benefit come from a full blown GUI rather then basic CLI output?

**Reply by fastlorenzo** [reply]
I think a CLI version should be sufficient for processing. The GUI part is more interesting if needing to put in reports.

**Reply by alexkornitzer** [reply]
Hmm, I wonder if we could wrap your work in Tauri, and bundle it into Chainsaw. No idea how much work that would be though, just a random though :)

**Reply by fastlorenzo** [reply]
Never used Tauri before, but I did try it out and made a quick PoC: https://github.com/fastlorenzo/rdpvis/tree/tauri-app :)
Still need some cleaning though, but I took the time as well to switch it to TypeScript

**Reply by fastlorenzo** [reply]
And with automated publication on GH Pages: https://fastlorenzo.github.io/rdpvis/ :smile:

**Reply by alexkornitzer** [reply]
Wicked, with that sort of structure it should be possible to embed chainsaw into it. I will have a play when I get some time.
